src/edu/csus/ecs/pc2/clics/API202306/CLICSClarification.java

```diff
@@ -0,0 +1,246 @@
+// Copyright (C) 1989-2024 PC2 Development Team: John Clevenger, Douglas Lane, Samir Ashoo, and Troy Boudreau.
+package edu.csus.ecs.pc2.clics.API202306;
+
+import java.util.ArrayList;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import edu.csus.ecs.pc2.core.Utilities;
+import edu.csus.ecs.pc2.core.log.Log;
+import edu.csus.ecs.pc2.core.log.StaticLog;
+import edu.csus.ecs.pc2.core.model.Clarification;
+import edu.csus.ecs.pc2.core.model.ClarificationAnswer;
+import edu.csus.ecs.pc2.core.model.ClientId;
+import edu.csus.ecs.pc2.core.model.ClientType;
+import edu.csus.ecs.pc2.core.model.ContestTime;
+import edu.csus.ecs.pc2.core.model.ElementId;
+import edu.csus.ecs.pc2.core.model.Group;
+import edu.csus.ecs.pc2.core.model.IInternalContest;
+import edu.csus.ecs.pc2.core.util.IJSONTool;
+import edu.csus.ecs.pc2.services.core.JSONUtilities;
+
+/**
+ * CLICS Clarification
+ * Contains information about a clarification
+ *
+ * @author John Buck
+ *
+ */
+public class CLICSClarification {
+
+    @JsonProperty
+    private String id;
+
+    @JsonProperty
+    private String from_team_id;
+
+    @JsonProperty
+    private String to_team_id;
+
+    @JsonProperty
+    private String [] to_team_ids;
+
+    @JsonProperty
+    private String [] to_group_ids;
+
+    @JsonProperty
+    private String reply_to_id;
+
+    @JsonProperty
+    private String problem_id;
+
+    @JsonProperty
+    private String text;
+
+    @JsonProperty
+    private String time;
+
+    @JsonProperty
+    private String contest_time;
+
+    @JsonProperty
+    private int number;
+
+    public CLICSClarification() {
+        // for jackson deserialize
+    }
+
+    public CLICSClarification(IInternalContest model, Clarification clar) {
+        this(model, clar, null);
+    }
+
+    /**
+     * Fills in the clarification properties
+     *
+     * @param model The contest
+     * @param clar The clarification
+     * @param clarAns non-null if this is an answer
+     */
+    public CLICSClarification(IInternalContest model, Clarification clar, ClarificationAnswer clarAns) {
+
+//        // use last answer if we were not handed the answer
```

Comment:
I assume this commented-out block is old (legacy) code?  I would suggest deleting it since the following code is a clear replacement for it and further this code is never going to be usable...

Reply:
Precisely.  I tried to share a single CLICSClarification object with a question and answer and that doesn't work for json serialization (object mapper). 

Code removed as suggested in next push.